src/main/java/at/xirado/bean/data/RankingSystem.java

```diff
@@ -101,7 +101,6 @@ public static long getTotalXPNeeded(int level) {
 
     public static long getTotalXP(@Nonnull Connection connection, long guildID, long userID) {
         var query = new SQLBuilder("SELECT totalXP FROM levels WHERE guildID = ? AND userID = ?")
-                .useConnection(connection)
                 .addParameters(guildID, userID);
         try (var result = query.executeQuery()) {
```

Comment:
Please remove all method overloads which allows passing in a connection, since you removed the functionality out of SQLBuilder